vpr/src/device/rr_graph_builder.h

```diff
@@ -185,15 +194,24 @@ class RRGraphBuilder {
         return node_storage_.count_rr_switches(num_arch_switches, arch_switch_inf, arch_switch_fanins);
     }
 
-    /** @brief This function reserve storage for RR nodes. */
+    /* Reserve the lists of nodes, edges, switches etc. to be memory efficient.
+     * This function is mainly used to reserve memory space inside RRGraph,
+     * when adding a large number of nodes/edge/switches/segments,
+     * in order to avoid memory fragements */
     inline void reserve_nodes(size_t size) {
         node_storage_.reserve(size);
     }
+    void reserve_segments(const size_t& num_segments);
```

Comment:
We can use ``size_t`` rather than ``const size_t&`` here, it is more memory efficient. 
Suggested by @hzeller , we use ``const &`` only for big data structures.